sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignature.java

```diff
@@ -149,14 +289,14 @@ static ProcessElementMethod create(
 
     /** Whether this {@link DoFn} uses a Single Window. */
     public boolean usesSingleWindow() {
-      return extraParameters().contains(Parameter.BOUNDED_WINDOW);
+      return extraParameters().contains(Parameter.boundedWindow());
```

Comment:
Also, Ben's other comments remind me that this could fairly be treated as just a shortcut. This needs to check for the presence of state and timer parameters as well, and might be most easily renamed to be its own converse `ignoresWindow`. Done now. (`Iterables.any` could be leverage `match` but actually it turns out quite verbose)